src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ChainedProcessListener.java

```diff
@@ -0,0 +1,21 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps;
+
+import org.geoserver.wps.executor.ProcessManager;
+
+/** An interface to monitor how processes are chained.
+ * 
+ *  @author etj (Emanuele Tajariol @ GeoSolutions)
+ */
+public interface ChainedProcessListener {
```

Comment:
Given the name, I was expecting to see notifications only for the chained processes, but it seems it's notifying also about the progress of the main one... we already have notifications for that one, couldn't a class implement both?
Also, since the interface is generic, why not notify about dismissal too?

Reply:
Javadoc updated to reflect for the fact that the listener is notified for both the root process and internal chained processes.
Modifying the existing listener would break its API contract.
This listener is a bit more "strict" about the subprocesses: it fires a return/fail when the process returns, while the existing listener lingers a bit on the output encoding.

Reply:
I see the updated javadoc on the ChainedProcessListenerNotifier (which is not public, good thanks) but not here in the public interface, which just states:

```An interface to monitor how processes are chained.```